x-pack/legacy/plugins/uptime/server/lib/adapters/framework/kibana_framework_adapter.ts

```diff
@@ -30,18 +29,46 @@ export class UMKibanaBackendFrameworkAdapter implements UMBackendFrameworkAdapte
   }
 
   public registerGraphQLEndpoint(routePath: string, schema: GraphQLSchema): void {
-    this.server.register<UMHapiGraphQLPluginOptions>({
-      options: {
-        graphQLOptions: (req: any) => ({
-          context: { req },
-          schema,
-        }),
-        path: routePath,
-        route: {
-          tags: ['access:uptime'],
-        },
+    const options = {
+      graphQLOptions: (req: any) => ({
+        context: { req },
+        schema,
+      }),
+      path: routePath,
+      route: {
+        tags: ['access:uptime'],
+      },
+    };
+    this.server.route({
+      options: options.route,
+      handler: async (request: Request, h: ResponseToolkit) => {
+        try {
+          const { method } = request;
+          const query =
+            method === 'post'
+              ? (request.payload as Record<string, any>)
+              : (request.query as Record<string, any>);
+
+          const graphQLResponse = await runHttpQuery([request], {
+            method: method.toUpperCase(),
+            options: options.graphQLOptions,
+            query,
+          });
+
+          return h.response(graphQLResponse).type('application/json');
+        } catch (error) {
+          if (error.isGraphQLError === true) {
+            return h
+              .response(error.message)
+              .code(error.statusCode)
+              .type('application/json');
+          }
+          return h.response(error).type('application/json');
+        }
       },
-      plugin: uptimeGraphQLHapiPlugin,
+      method: ['get', 'post'],
```

Comment:
This is a useful example for other teams for how to replace the hapi plugin. When you switch to using the NP router in your shim (from `server.newPlatform.setup.core.http.createRouter()`) you won't be able to define one route with two methods (get & post) so this will have to be split into two routes.